### ISSUE TITLE
compute_viewshed: improve code and description

### ISSUE THREAD

**Issue by spono** [land-info-lab/viewscape#2]
Hi,
I would suggest to improve the description of the `r` parameter in `compute_viewshed`. Actually it reports _"it is defaulted as NULL"_ but this does not explain why, in the code, `NULL` leads to a fixed distance of 1000 m / 3281 ft.

IMO, it could be better to leave `r` as a numeric parameter that -BTW- will be multiplied to whatever the unit of the layer is. According to this, I would remove the following lines

```
  dsm_units <- sf::st_crs(dsm)$units
  if(is.null(r) == TRUE){
    if (dsm_units == "ft") {
      r <- 3281
    } else if (dsm_units == "m") {
      r <- 1000
    }
  }
```

and change them with a check for CRS of the layers:

`if( is.na(sf::st_crs(viewpoints)) | is.na(sf::st_crs(dsm)) ){ stop("One or more layers are missing the CRS") }`

**Comment by billbillbilly** [comment]
Thanks for your suggestion! The description will be updated on this git. It may be a good idea to check CRS of the layers but viewpoints could be not only sf object but also a vector of coordinates (projected). I think it may make sense to apply it when the input viewpoints is sf. What do you think @spono ?

**Comment by spono** [comment]
> it may make sense to apply it when the input viewpoints is sf

I agree even though I would suggest to check [this](https://stackoverflow.com/questions/18295825/determine-if-point-is-within-bounding-box) to include also a check on coordinates and create a full solution.

**Comment by billbillbilly** [comment]
Thanks for sharing this. I will add a check on coordinates to determine if the input viewpoint(s) is within the extent of the input raster.
